org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterMeasurementPower.java

```diff
@@ -145,7 +145,7 @@ public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
         return ChannelBuilder
                 .create(createChannelUID(thingUID, endpoint,
                         ZigBeeBindingConstants.CHANNEL_NAME_ELECTRICAL_ACTIVEPOWER),
-                        ZigBeeBindingConstants.ITEM_TYPE_NUMBER)
+                        ZigBeeBindingConstants.ITEM_TYPE_NUMBER_POWER)
```

Comment:
I'm assuming when we dynamically generate channels, e.g. for a generic device without a thing XML definition, we ignore the contents of `channels.xml` completely? Thus without changing the type here we were not actually getting the correct channel type?

Reply:
No - the channels XML file is not ignored. The channel type definitions are still required. When you define a channel (either programatically or in the thing XML file) you have to specify the channel-type, but the channel can also override certain aspects of this.  I'm pretty sure that if you try and define a channel that doesn't have a channel type defined, it will error.